app/models/repo.js

```diff
@@ -68,7 +69,23 @@ const Repo = VcsEntity.extend({
     return hasPermissions && (!isMigrated || isFailed);
   }),
 
-  allowance: reads('owner.allowance'),
+  allowance: computed('owner.allowance', 'repoOwnerAllowance', function () {
+    if (this.owner.allowance)
+      return this.owner.allowance;
+    if (this.repoOwnerAllowance)
+      return this.repoOwnerAllowance;
+    this.fetchRepoOwnerAllowance.perform();
+  }),
+
+  repoOwnerAllowance: reads('fetchRepoOwnerAllowance.lastSuccessful.value'),
+
+  fetchRepoOwnerAllowance: task(function* () {
+    const allowance = this.store.peekRecord('allowance', this.owner.id);
```

Comment:
`peekRecord` takes a model name and record id as arguments. And id assumed to be of the same type as the model name. Therefore, `peekRecord` with this implementation will return an `allowance` record, which `id` is equal to `this.owner.id`, which seems wrong to me. `owner` and `allowance` have different `id`s, aren't they? 
According to the current data structures, I'd suggest the following:
```suggestion
    const allowance = this.store.peekAll('allowance ').findBy('owner.id', this.owner.id);
```

Reply:
Api provides: allowance.id === owner.id
so should be right :-)